low-level/sensors/impl/platforms/realstor/realstor_disk_sensor.py

```diff
@@ -271,6 +271,17 @@ def _rss_check_disks_presence(self):
             #raise alert for added drive
             self._rss_raise_disk_alert(self.rssencl.FRU_INSERTION, disk_info)
 
+            # Update health status for inserted disk in memfault cache,
+            # to raise fault alert after insertion if inserted disk status is not OK.
+            if disk_info["health"] != "OK":
+                for id_fault, cached_fault in enumerate(self.rssencl.memcache_faults):
+                    #fetch disk slot from component_id present in memcache_faults.
+                    component_id = cached_fault["component-id"]
+                    if component_id.startswith('Disk 0'):
```

Comment:
Why do this only for 'Disk 0' ?

Reply:
It is not Disk 0 specific. from mc api response we get disk number in component-id and which will start with Disk 0.
Example. Disk 0.3, Disk 0.45, Disk 0.80 etc.
We have added this filter because along with Disk we also get information of other unhelthy components i.e controller, fan, disk group etc.

Reply:
Any disk component-id represented as  'Disk 0.<numeral> like 'Disk 0.7', 'Disk 0.80' etc
So 'Disk 0' is used as common regex. Other resource which has similarity in component-id is 'Disk Group' , so 'Disk 0' used in code above to catch only disks.